### PR TITLE
Update release note for 2.14 release

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description
Update release note for 2.14 release

### Issues Resolved
Update release note for 2.14 release

### Check List
- [ ] New functionality includes testing.
    - [ ] All tests pass
- [ ] New functionality has been documented.
    - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Features
 ### Enhancements
 ### Bug Fixes
-- Fix multi node "no such index" error in text chunking processor ([#713](https://github.com/opensearch-project/neural-search/pull/713))
 ### Infrastructure
 ### Documentation
+- Include latest change in release notes ([#734](https://github.com/opensearch-project/neural-search/pull/734))
```
Comment (jmazanec15) [inline CHANGELOG.md:21]: Lets remove this - I dont get much information from the note. I think it can be skipped.

Comment (yuye-aws) [reply CHANGELOG.md:21]: Sure. Will removing this cause change log failure in github checks?

**Review by jmazanec15 — APPROVED** [review]
LGTM thanks

**Comment by yuye-aws** [comment]
@jmazanec15 Need to also attach backport 2.x label.

**Review by zhichao-aws — APPROVED** [review]

**Pull request merged**